content/articles/statistical-hypothesis-testing-python-implementation/index.md

```diff
@@ -0,0 +1,519 @@
+sm.stats.anova_lm(model, typ=2)
+```
+![twowayanova](twowayanova.png)
+
+The P-value of petal width is greater than 0.05, which increases the probability of our null hypothesis being accepted. 
+
+The P-value of sepal length is less than 0.05, which increases the probability of our null hypothesis being rejected.
+
+Likewise, the P-value of both petal_width and sepal_length is less than 0.05, we reject the null hypothesis because it is obvious one or more samples are very much different in their means. That is to say, the average of all data samples is not equal.
+
+### Conclusion
+At the end of this tutorial, I believe the readers should have more than a grasp of the concept of statistical hypothesis testing and how to implement it with python programming. To carry out hypothesis testing, the first step is to form an initial assumption and label it as H0. The next step would be to collect all data samples available to support our hypothesis, collect all the shreds of evidence and analyze the data, and make a decision whether to accept the H0 or reject it. However, errors of type 1 and type 2 have been explained using an assumed pregnant woman as an example, which is something almost everyone can relate with. When we reject the ground truth but eventually it is true, we encounter the *Type 1* error. On the flip side, when we do not reject the ground truth despite the fact that it is not true, we encounter the *Type 2* error.
+
+Additional resources are added below to better solidify the knowledge gained from this tutorial.
+
+**Happy coding**
+
+### Extra Resources
+1. [Hypothesis Testing in Data Science](https://www.section.io/engineering-education/hypothesis-testing-data-science/)
+2. [Chi Square Test](https://byjus.com/maths/chi-square-test/)
+3. [Understanding ANOVA test](https://www.qualtrics.com/experience-management/research/anova/)
+4. [ANOVA Test: Definition, Types, Examples](https://www.statisticshowto.com/probability-and-statistics/hypothesis-testing/anova/)
+5. [Understanding t-Tests: t-values and t-distributions](https://blog.minitab.com/en/adventures-in-statistics-2/understanding-t-tests-t-values-and-t-distributions)
+6. [MAKING SENSE OF THE TWO-SAMPLE T-TEST](https://www.isixsigma.com/tools-templates/hypothesis-testing/making-sense-two-sample-t-test/)
+7. [Understanding Hypothesis Tests: Significance Levels (Alpha) and P values in Statistics](https://blog.minitab.com/en/adventures-in-statistics-2/understanding-hypothesis-tests-significance-levels-alpha-and-p-values-in-statistics)
+8. [Chi-squared test application](http://www.openanesthesia.org/chi-squared_test_application/)
+9. [Chi-square mathematics for statistics students](https://www.yourarticlelibrary.com/project-reports/chi-square-test/chi-square-test-meaning-applications-and-uses-statistics/92394).
+10. [What are mutually exclusive events](https://www.mathsisfun.com/data/probability-events-mutually-exclusive.html)
```

Comment:
Please add resources that aren't already mentioned in the article, and make sure that the resources are articles or papers that can further enhance the knowledge of the reader and not basic ones.

Reply:
done